src/senaite/core/browser/samples/view.py

```diff
@@ -661,6 +665,17 @@ def add_custom_transitions(self):
         if copy_to_new:
             custom_transitions.append(copy_to_new)
 
+        # Allow to create a worksheet for the selected samples
+        if self.can_create_worksheet():
+            custom_transitions.append({
+                "id": "modal_create_worksheet",
+                "title": _("Create Worksheet"),
+                "url": "{}/create_worksheet_modal".format(
+                    api.get_url(self.context)),
+                "css_class": "btn btn-outline-secondary",
+                "help": _("Create a new worksheet for the selected samples")
+            })
+
```

Comment:
Better to only display this transition on "Active" and "Received" review statuses/filters. It does not make sense for e.g. verified or published samples